File: src/ERP/acme.atena.domain/DTO/Seguranca/Tela.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace acme.atena.domain.DTO.Seguranca
{
    public class Tela: AbstractEntity
    {
        public string Name { get; set; }
        public string Url { get; set;}
        public string Icon { get; set;}
        public bool Title { get; set; } = false;
        public bool IsPrincipal { get; set; }
        public string Class { get; set; }
        public string Variant { get; set; }

        public Guid TelaSistemaFilhaId { get; set; }

        public Tela TelaSistemaFilha { get; set; }

        public ICollection<Tela> TelasSistemasFilhas { get; set; } = new HashSet<Tela>();
        public ICollection<PermissaoUsuario> PermissaoUsuarios { get; set; } = new HashSet<PermissaoUsuario>();

    }
}

File: site/coreui-free-angular-admin-template-master/src/app/view-model/response/security/tela.ts
import { AbstractEntity } from "../abstract-entity";
import { PermissaoUsuario } from "./permissao-usuario";

export class Tela extends AbstractEntity{
    name:string ;
    url:string;
    icon:string;
    title:boolean ;
    isPrincipal :boolean;
    class:string  ;
    variant:string  ;

    telaSistemaFilhaId:string;

    telaSistemaFilha:Tela;

    telasSistemasFilhas:Tela[] = [];
    permissaoUsuarios:PermissaoUsuario[]  = [];

}
